File: daily/2021/12/1206_P12977.py
# 2021-12-06 프로그래머스 Level01 소수 만들기
from itertools import combinations


def is_prime(x):
    for i in range(2, x):
        if x % i == 0:
            return False
    return True


def solution(a):

    ans = 0
    arr = list(combinations(a, 3))
    print(arr)
    for n1, n2, n3 in arr:
        if is_prime(n1+n2+n3):
            ans += 1
    return ans


a = [1, 2, 3, 4]
b = [1, 2, 7, 6, 4]
print(solution(b))

File: daily/2021/12/1206_P12977.java

class Solution {
    public static void main(String[] args) {
        int[] arr = { 1, 2, 3, 4 };
        int ans = solution(arr);
        System.out.println(ans);
    }

    public static int solution(int[] nums) {
        int ans = 0;

        for (int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                for (int k = j + 1; k < nums.length; k++) {
                    int num = nums[i] + nums[j] + nums[k];
                    if (sosu(num)) {
                        ans++;
                    }
                }
            }
        }
        return ans;
    }

    public static boolean sosu(int num) {
        boolean chk = true;
        if (num == 2) {
            return chk;
        }
        for (int i = 2; i < num; i++) {
            if (num % i == 0) {
                chk = false;
                break;
            }
        }
        return chk;
    }

}